crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -929,6 +929,133 @@ impl Measured2d for Annulus {
     }
 }
 
+/// A rhombus primitive, also known as a diamond shape.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Diamond")]
+pub struct Rhombus {
+    /// Size of the horizontal and vertical diagonals of the rhombus
+    pub half_diagonals: Vec2,
+}
+impl Primitive2d for Rhombus {}
+
+impl Default for Rhombus {
+    /// Returns the default [`Rhombus`] with a half-horizontal and half-vertical diagonal of `0.5`.
+    fn default() -> Self {
+        Self {
+            half_diagonals: Vec2::splat(0.5),
+        }
+    }
+}
+
+impl Rhombus {
+    /// Create a new `Rhombus` from a vertical and horizontal diagonal sizes.
+    #[inline(always)]
+    pub fn new(horizontal_diagonal: f32, vertical_diagonal: f32) -> Self {
+        Self {
+            half_diagonals: Vec2::new(horizontal_diagonal / 2.0, vertical_diagonal / 2.0),
+        }
+    }
+
+    /// Create a new `Rhombus` from a side length with all inner angles equal.
+    #[inline(always)]
+    pub fn from_side(side: f32) -> Self {
+        Self {
+            half_diagonals: Vec2::new(side.hypot(side) / 2.0, side.hypot(side) / 2.0),
+        }
+    }
+
+    /// Create a new `Rhombus` from a given inradius with all inner angles equal.
+    #[inline(always)]
+    pub fn from_inradius(inradius: f32) -> Self {
+        let half_diagonal = inradius * 2.0 / std::f32::consts::SQRT_2;
+        Self {
+            half_diagonals: Vec2::new(half_diagonal, half_diagonal),
+        }
+    }
+
+    /// Get the length of each side of the rhombus
+    #[inline(always)]
+    pub fn side(&self) -> f32 {
+        self.half_diagonals.x.hypot(self.half_diagonals.y)
+    }
+
+    /// Get the radius of the circumcircle on which all vertices
+    /// of the rhombus lie
+    #[inline(always)]
+    pub fn circumradius(&self) -> f32 {
+        self.half_diagonals.x.max(self.half_diagonals.y)
+    }
+
+    /// Get the radius of the largest circle that can
+    /// be drawn within the rhombus
+    #[inline(always)]
+    #[doc(alias = "apothem")]
+    pub fn inradius(&self) -> f32 {
+        ((self.half_diagonals.x * self.half_diagonals.y) / self.side()).max(0.0)
```

Comment:
This resolves the issue with the division by 0? Didn't know that's possible.

Reply:
`x / 0.0` is positive infinity in Rust iirc, so I'm pretty sure this does *not* work

Reply:
When testing, the result of the division by 0 appeared as `NaN` to us, and when we used this method it was working correctly. But it really is a little bit weird so we changed the function and added the `if` statement, in our latest commit.